core-to-plc/src/Language/Plutus/CoreToPLC/Types.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE ConstraintKinds  #-}
+{-# LANGUAGE FlexibleContexts #-}
+
+module Language.Plutus.CoreToPLC.Types where
+
+import           Language.Plutus.CoreToPLC.Error
+
+import qualified Language.PlutusCore             as PLC
+import           Language.PlutusCore.Quote
+
+import qualified GhcPlugins                      as GHC
+
+import           Control.Monad.Except
+import           Control.Monad.Reader
+import           Control.Monad.State
+
+import qualified Data.List.NonEmpty              as NE
+import qualified Data.Map                        as Map
+import qualified Data.Text                       as T
+
+type PLCExpr = PLC.Term PLC.TyName PLC.Name ()
+type PLCType = PLC.Type PLC.TyName ()
+
+type ConvError = WithContext T.Text (Error ())
+
+type PrimTerms = Map.Map GHC.Name (Quote PLCExpr)
+type PrimTypes = Map.Map GHC.Name (Quote PLCType)
+
+newtype ConversionOptions = ConversionOptions { coCheckValueRestriction :: Bool }
+
+type ConvertingContext = (ConversionOptions, GHC.DynFlags, PrimTerms, PrimTypes, ScopeStack)
```

Comment:
I'd prefer a `data`.
(This module is exactly how a core module with data types should look like)

Reply:
Fair - but this is not really part of this PR, so I'll do it in a follow up. (This stuff just move)

Reply:
Eh, actually, I'll put it in here.